src/interaction/Simulation.cpp

```diff
@@ -0,0 +1,183 @@
+#include "../include/interaction/Simulation.h"
+#include "../include/vrapp/VRVolumeApp.h"
+
+Simulation::Simulation(VRVolumeApp &controller_app, float time) : m_animation_duration(time), m_controller_app(controller_app),
+                                                                  animation_button_label("ANIMATE"), m_animation_state(STOP)
+{
+}
+
+void Simulation::add_simulation_state(SimulationState &simulationState)
+{
+  float time = m_simulation_states.size() * m_animation_duration;
+
+  int mins = (int)time / 60;
+  int seconds = (int)time % 60;
+
+  simulationState.time_label = std::to_string(mins) + ":" + std::to_string(seconds);
+  m_simulation_states.push_back(simulationState);
+}
+
+void Simulation::create_animations()
+{
+  if (m_simulation_states.size() > 1)
+  {
+    SimulationState first_position = m_simulation_states.front();
+    ch::Sequence<glm::vec3> sequence_eye(first_position.poi.eye);
+    ch::Sequence<glm::vec3> sequence_target(first_position.poi.target);
+    ch::Sequence<glm::vec3> sequence_up(first_position.poi.up);
+    ch::Sequence<float> sequence_radius(first_position.poi.radius);
+    ch::Sequence<glm::vec3> sequence_clip_max(first_position.max_clip);
+    ch::Sequence<glm::vec3> sequence_clip_min(first_position.min_clip);
+
+    for (auto iterator = std::next(m_simulation_states.begin()); iterator != m_simulation_states.end(); iterator++)
```

Comment:
all the other for loops use the `i; i < size, ++i` style - is there a reason this one uses the iterator style instead?